File: solutions/Binary-Tree-Level-Order-Traversal-II/Binary-Tree-Level-Order-Traversal-II.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrderBottom = function (root) {
  const ans = [];
  if (root === null) {
    return ans;
  }
  let levels = [root];
  while (levels.length > 0) {
    const temp = [];
    const curLevel = [];
    while (levels.length > 0) {
      const current = levels.pop();
      curLevel.push(current.val);
      if (current.left !== null) {
        temp.unshift(current.left);
      }
      if (current.right !== null) {
        temp.unshift(current.right);
      }
    }
    ans.unshift(curLevel);
    levels = temp;
  }
  return ans;
};

File: solutions/Binary-Tree-Level-Order-Traversal-II/Binary-Tree-Level-Order-Traversal-II.cpp

/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int> > levelOrderBottom(TreeNode *root) {
        vector<vector<int> >ans;
        if(!root)
            return ans;
        queue<vector<TreeNode*> >q;
        vector<TreeNode*>vec;
        vec.push_back(root);
        q.push(vec);
        while(!q.empty())
        {
            vec=q.front();
            q.pop();
            vector<TreeNode*>temp;
            vector<int>t;
            for(int i=0;i<vec.size();i++)
            {
                t.push_back(vec[i]->val);
                if(vec[i]->left)
                    temp.push_back(vec[i]->left);
                if(vec[i]->right)
                    temp.push_back(vec[i]->right);
            }
            if(!t.empty())
                ans.push_back(t);
            if(!temp.empty())
                q.push(temp);
        }
        reverse(ans.begin(),ans.end());
        return ans;
    }
};


 